orm/encoding/ormvalue/codec.go

```diff
@@ -0,0 +1,103 @@
+package ormvalue
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/cosmos/cosmos-sdk/orm/types/ormerrors"
+
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+// Codec defines an interface for decoding and encoding values in ORM index keys.
+type Codec interface {
+
+	// Decode decodes a value in a key.
+	Decode(r *bytes.Reader) (protoreflect.Value, error)
```

Comment:
Why do we prefer a concrete type here instead of an interface, it creates dissonance with Encode IMO.

Reply:
I added an interface to address